### PR TITLE
RDM-1831 Hide and show condition for a field on a case view tab and case view tab field.

### PR DESCRIPTION
**Pull request by andrewfolga**
Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-1831
https://tools.hmcts.net/jira/browse/RDM-1591
https://tools.hmcts.net/jira/browse/RDM-1964

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseViewField.java [hunk 1]
```diff
@@ -121,4 +122,10 @@ public Boolean getShowSummaryChangeOption() {
     public void setShowSummaryChangeOption(final Boolean showSummaryChangeOption) {
         this.showSummaryChangeOption = showSummaryChangeOption;
     }
+
+    @Override
+    public String toString() {
+        return ReflectionToStringBuilder.toString(this);
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseViewField.java]: beware of using reflection. Can slow down performance. Perhaps an IDE generated toString() better?

File: src/test/resources/mappings/display-tab-structure.json [hunk 2]
```diff
@@ -15,7 +15,9 @@
         "id" : "NameTab",
         "label" : "Name",
         "order" : 1,
+        "show_condition": "NameTabShowCondition",
         "tab_fields" : [{
+          "show_condition": "PersonFirstNameFieldShowCondition",
```
Comment (mario-paniccia) [inline src/test/resources/mappings/display-tab-structure.json]: perhaps we can add a valid example show condition here?

File: src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseViewField.java (hunk 1 above)
Comment (andrewfolga) [reply src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseViewField.java]: Yeah maybe we should agree on a standard here. I can see many different usages across the code base including:
- StringBuilder,
- ReflectionToStringBuilder,
- commons-lang ToStringBuilder,
- commons-lang3 ToStringBuilder

I would lean towards the last option. What are your thoughts?

Comment (andrewfolga) [reply src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseViewField.java]: ok changed to commons-lang3 ToStringBuilder

**Pull request merged**